### PR TITLE
Add new easyconfig files for OpenFOAM and OpenFOAM-Extend

### PR DESCRIPTION
**Pull request by besserox**
This PR brings these 3 new easyconfig files and their dependencies:
- OpenFOAM-2.0.1-foss2016a + a patch to fix compilation error
- OpenFOAM-Extend-3.2-foss-2016a
- OpenFOAM-Extend-3.2-ictce-7.3.5

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@besserox PRs with more than 10 easyconfigs tend to be a pain to get merged, but let's see how much of a problem this ends up being, since I suspects this is mostly a toolchain bump

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.0.1-foss-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+source_urls = ['http://downloads.sourceforge.net/foam/%(version)s']
+sources = [
+    'OpenFOAM-%(version)s.gtgz',
+    'ThirdParty-%(version)s.gtgz',
+]
+
+patches = [
+    'cleanup-OpenFOAM-%(version)s.patch',
+    'cleanup-OpenFOAM-2.0.1-GCC49.patch',
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.0.1-foss-2016a.eb]: when adding new patches, please name them `OpenFOAM-%(version)s_<label>.patch`

existing patches for older versions of OpenFOAM (as is the case here) can be left as-is

File: easybuild/easyconfigs/o/OpenFOAM-Extend/OpenFOAM-Extend-3.2-ictce-7.3.5.eb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+patches = [
+    'OpenFOAM-Extend-%(version)s-ParMGridGen.patch',
+    'OpenFOAM-Extend-3.1_build-qa.patch',
+    'OpenFOAM-Extend-3.1_comp-mpi.patch',
+    'OpenFOAM-Extend-3.1_skip-ThirdParty-OpenMPI.patch',
+]
+
+dependencies = [
+    ('ParMETIS', '4.0.3'),
+    ('METIS', '5.1.0'),  # order matters, METIS need to be listed after ParMETIS to get $CPATH right
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM-Extend/OpenFOAM-Extend-3.2-ictce-7.3.5.eb]: existing OpenFOAM-Extend 3.2 easyconfigs use the 32bit variant of METIS

File: easybuild/easyconfigs/o/OpenFOAM-Extend/OpenFOAM-Extend-3.2-foss-2016a.eb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+patches = [
+    'OpenFOAM-Extend-%(version)s-ParMGridGen.patch',
+    'OpenFOAM-Extend-3.1_build-qa.patch',
+    'OpenFOAM-Extend-3.1_comp-mpi.patch',
+    'OpenFOAM-Extend-3.1_skip-ThirdParty-OpenMPI.patch',
+]
+
+dependencies = [
+    ('ParMETIS', '4.0.3'),
+    ('METIS', '5.1.0'),  # order matters, METIS need to be listed after ParMETIS to get $CPATH right
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM-Extend/OpenFOAM-Extend-3.2-foss-2016a.eb]: existing OpenFOAM-Extend 3.2 easyconfigs use the 32bit variant of METIS

**Comment by boegel** [comment]
Some minor remarks, pretty much looks like just a toolchain bump, so probably doesn't require splitting up.

**Comment by besserox** [comment]
@boegel It should be fixed now.

Yes, all these easyconfigs are toolchain bumps for ictce/7.3.5 and foss/2016a

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 30 out of 30 (18 easyconfigs in this PR)
node2635.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/e8ae9e501ddc0a563623 for a full test report.

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.0.1-foss-2016a.eb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+source_urls = ['http://downloads.sourceforge.net/foam/%(version)s']
+sources = [
+    'OpenFOAM-%(version)s.gtgz',
+    'ThirdParty-%(version)s.gtgz',
+]
+
+patches = [
+    'cleanup-OpenFOAM-%(version)s.patch',
+    ' OpenFOAM-%(version)s-fix_compilation_errors_GCC49.patch
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-2.0.1-foss-2016a.eb]: missing end `'`, and space in front that you probably don't want :)

**Comment by besserox** [comment]
@boegel I got lost among the dependencies while fixing it. But it ok now. I have fixed your initial comments. To summarize: 

It is just 3 new easyconfigs
- OpenFOAM-2.0.1-foss2016a + a patch to fix compilation error
- OpenFOAM-Extend-3.2-foss-2016a
- OpenFOAM-Extend-3.2-ictce-7.3.5

The other easyconfigs are toolchain bumps for ictce/7.3.5 and foss/2016a of the dependencies.

**Comment by boegel** [comment]
@besserox: if you feel this is ready, please update the PR title and drop in a comment;

I'll submit another test report for this, just for kicks. :)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 20 out of 20 (20 easyconfigs in this PR)
node2699.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5174c6434044b81358aa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 19 out of 20 (20 easyconfigs in this PR)
node2036.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f772baa6f41c94d514af for a full test report.

**Comment by besserox** [comment]
@boegel Is that possible to get the full output of the failed build at https://gist.github.com/boegel/f772baa6f41c94d514af ?

**Comment by boegel** [comment]
@besserox Sorry for not getting back to you on this... Did you mean to close it, or should we still try and get it in?

**Comment by besserox** [comment]
@boegel I don't need those anymore. So I prefer not to spend time on it.

**Pull request closed without merging**